File: PRACTICA#2/COMPOSICION-AGREGACION/1/JAVA/Casa.java
import java.util.ArrayList;
import java.util.List;

public class Casa {
    private String dirección;
    private List<Habitacion> habitaciones;

    public Casa(String dirección) {
        this.dirección = dirección;
        this.habitaciones = new ArrayList<>();
    }

    //A
    public String getDirección() {
        return dirección;
    }

    public void setDirección(String dirección) {
        this.dirección = dirección;
    }

    public List<Habitacion> getHabitaciones() {
        return habitaciones;
    }

    public void setHabitaciones(List<Habitacion> habitaciones) {
        this.habitaciones = habitaciones;
    }

    public void agregar_habitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrar_casa() {
        System.out.println("Dirección: " + dirección);
        System.out.println("Habitaciones:");
        for (Habitacion habitacion : habitaciones) {
            habitacion.mostrar_info();
        }
    }
}

File: PRACTICA#2/COMPOSICION-AGREGACION/1/PYTHON/Casa.py
class Casa:
    def __init__(self, dirección):
        self.__dirección = dirección
        self.__habitaciones = []

    #A
    def get_dirección(self):
        return self.__dirección

    def set_dirección(self, dirección):
        self.__dirección = dirección

    def get_habitaciones(self):
        return self.__habitaciones

    def set_habitaciones(self, habitaciones):
        self.__habitaciones = habitaciones

    def agregar_habitacion(self, habitacion):
        self.__habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"Dirección: {self.__dirección}")
        print("Habitaciones:")
        for habitacion in self.__habitaciones:
            habitacion.mostrar_info()